lib/scenario_runner.py

```diff
@@ -369,7 +371,7 @@ def merge_dicts(dict1, dict2):
                         else:
                             dict1[k] = v
                     return dict1
-                return dict1
+                return dict2
```

Comment:
Is this on purpose? Because we set the new keys in `dict1`

Reply:
I think this is the bug in the code, this is why I pinged you to look at it.

To my understanding this branch of the recursion is a terminal condition and will only be run if a string key is encountered that has no children. In that case you want to return the key from the usage_scenario.yml

In the other branches you are merging data in from the compose.yml in case the key exists in both.

I hope I understood your code correctly. Please give it a detailed look again

Reply:
Yes. I agree that it looks like an error. Good catch